### PR TITLE
Remove the parts of multiple rank function calls

### LINKED ISSUE #1581: Update supported TopN pattern
> After this bug fixed, https://docs.risingwave.com/docs/current/sql-pattern-topn/ should be updated. Top-N pattern with multiple rank function calls will not be supported any more.

_Originally posted by @stdrc in https://github.com/risingwavelabs/risingwave/issues/13653#issuecomment-1832206619_

### PR DESCRIPTION
**Pull request by WanYixian**
## Info

- **Description**

  Top-N pattern with multiple rank function calls are not supported anymore, so remove the description and syntax about that in our doc.

- **Related code PR**

  https://github.com/risingwavelabs/risingwave/issues/13653

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/1581

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/sql/syntax/sql-pattern-topn.md [hunk 1]
```diff
@@ -74,10 +73,8 @@ SELECT r1
   FROM (
     SELECT
       *,
-      row_number() OVER (PARTITION BY x ORDER BY y) r1,
-      row_number() OVER (PARTITION BY x, y ORDER BY z) r2,
-      rank() OVER (PARTITION BY x, y ORDER BY z) r3
+      row_number() OVER (PARTITION BY x ORDER BY y) r,
```
Comment (stdrc) [inline docs/sql/syntax/sql-pattern-topn.md]:
```suggestion
      row_number() OVER (PARTITION BY x ORDER BY y) r
```

File: docs/sql/syntax/sql-pattern-topn.md [hunk 2]
```diff
@@ -47,7 +46,7 @@ You must follow the pattern exactly to construct a valid Top-N query.
...
 |`PARTITION BY` clause |Specifies the partition columns. Each partition will have a Top-N result.|
 |`ORDER BY` clause|Specifies how the rows are ordered.|
-|rank_range|Specifies the range of the rank number. The rank range is required for the query to be recognized as a top-N query. If multiple ranking function clauses are included, you can include multiple rank filtering conditions as the rank range. For example, `rank1 < 5 AND rank2 < 3`. The range can be specified in other forms. <br />Examples: `WHERE M < rank AND rank < N` or `WHERE rank between M and N`. Optionally, you can specify any additional conditions to further filter the results. |
+|rank_range|Specifies the range of the rank number. The rank range is required for the query to be recognized as a top-N query. Optionally, you can specify any additional conditions to further filter the results. |
```
Comment (stdrc) [inline docs/sql/syntax/sql-pattern-topn.md]:
```suggestion
|rank_range|Specifies the range of the rank number. The rank range is required for the query to be recognized as a top-N query. The range can be specified in other forms. <br />Examples: `WHERE M < rank AND rank < N` or `WHERE rank between M and N`. Optionally, you can specify any additional conditions to further filter the results. |
```

**Review by stdrc — APPROVED** [review]

File: docs/sql/syntax/sql-pattern-topn.md (hunk 1 above)
Comment (WanYixian) [reply docs/sql/syntax/sql-pattern-topn.md]: Revised.

File: docs/sql/syntax/sql-pattern-topn.md (hunk 2 above)
Comment (WanYixian) [reply docs/sql/syntax/sql-pattern-topn.md]: Add this sentence now.

**Review by hengm3467 — APPROVED** [review]
LGTM. Thanks for the solid work.

**Pull request merged**